### ISSUE TITLE
documentation for no-return-await incorrect

### ISSUE THREAD

**Issue by ysangkok** [eslint/eslint#8276]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.2
* **npm Version:** n/a

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

```




```

**What did you do? Please include the actual source code causing the issue.**

```js
const request = require('then-request');

async function annotateRejection(name, promiseToAnnotate) {
    try {
        return await promiseToAnnotate; // eslint-disable-line no-return-await
    } catch (e) {
        e.message = name + ': ' + e.message;
        throw e;
    }
}

module.exports = async (req /* : express$Request */, res /* : express$Response */) => {
    const options = {timeout: 3000};

    let replies;

    try {
        const req1Promise = annotateRejection('req1', request('GET', url1, options));
        const req2Promise = annotateRejection('req2', request('GET', url2, options));
        const req3Promise = annotateRejection('req3', request('GET', url3, options));

        replies = await Promise.all([req1Promise, req2Promise, req3Promise]);
    } catch (e) {
        console.error(e.stack);
    }
}
```

**What did you expect to happen?**

I have a use case that is not in either of these buckets:
https://github.com/eslint/eslint/commit/66fe9ffa72487343fd2e21673b79281424cff6ef#diff-a53fde9ed9249a9bbb8e54cea68ebb9eR37

If I do not `return await`, the exception handler in the annotator does not work (I am testing with timeouts). If I do not annotate, I can't see where the exception comes from.

See example code above.

**What actually happened? Please include the actual, raw output from ESLint.**

"No return await lint" documentation is inaccurate, makes return await worse than it is.

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, the `no-return-await` rule doesn't actually report this code due to the `try` statement, so I don't think you would need to disable the rule to allow this pattern.

**Comment by alberto** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the ESLint Gitter if you have any other issues!
